beta-20170406.md

```diff
@@ -0,0 +1,76 @@
+- The [`BYTES`](bytes.html) type name is now reported as `bytea` in `pg_catalog.pg_type` for compatibility with PostgreSQL. [#14495](https://github.com/cockroachdb/cockroach/pull/14495)
+- The PostgreSQL type IO functions (like `int8in`, `int8out`, `int8send`, and `int8recv`) have dummy implementations for compatibility with some ORMs. [#14529](https://github.com/cockroachdb/cockroach/pull/14529)
+
+### Command-Line Interface Changes
+
+- Command history no longer corrupts previous commands when normalizing them. [#14496](https://github.com/cockroachdb/cockroach/pull/14496)
+- The [`cockroach start`](start-a-node.html) no longer uses the `COCKROACH_PORT` environment variable; the port must be set on the server's command line (similar to `--host` and other flags). [#14612](https://github.com/cockroachdb/cockroach/pull/14612)
+
+### Admin UI Changes
+
+- The **Node List** page has been improved. [#14395](https://github.com/cockroachdb/cockroach/pull/14395)
+- A new `node-id` metric with labels for the node's address is now exported for use in external monitoring systems. [#14544](https://github.com/cockroachdb/cockroach/pull/14544)
+
+### Performance Improvements
+
+- A single `TimestampCache` is now shared by all replicas on a store, providing better control over memory usage and better support for large transactions. [#14489](https://github.com/cockroachdb/cockroach/pull/14489)
+- Reduced memory used by the `TimestampCache`. [#14516](https://github.com/cockroachdb/cockroach/pull/14516) [#14514](https://github.com/cockroachdb/cockroach/pull/14514)
+- Replica garbage collection is now much more efficient. [#14391](https://github.com/cockroachdb/cockroach/pull/14391)
+- [Adding](add-column.html) and [dropping](drop-column.html) columns now uses the distributed SQL engine. [#14331](https://github.com/cockroachdb/cockroach/pull/14331)
+
+### Bug Fixes
+
+- [`INSERT ON CONFLICT DO UPDATE`](insert.html#update-values-on-conflict) no longer uses an error-prone optimization. The [`UPSERT`](upsert.html) statement is now faster than the equivalent `INSERT ON CONFLICT DO UPDATE` statement. [#14485](https://github.com/cockroachdb/cockroach/pull/14485)
+- Joining nodes of two different clusters is now detected more reliably. [#14475](https://github.com/cockroachdb/cockroach/pull/14475)
+- Fixed a panic during prepared [`UPDATE`](update.html) statements. [#14481](https://github.com/cockroachdb/cockroach/pull/14481)
+- [`FLOAT`](float.html) and [`DECIMAL`](decimal.html) values in primary keys and indexes now consider `0` and `-0` equal (although the sign will correctly be retained). [#14528](https://github.com/cockroachdb/cockroach/pull/14528)
+- [`DECIMAL`](decimal.html) values in primary keys and indexes no longer have zeros truncated. [#14528](https://github.com/cockroachdb/cockroach/pull/14528)
```

Comment:
At the end I'd add:

Secondary indexes where the original value had trailing zeros can be dropped and recreated to correctly return them when that index is used.